src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -526,22 +648,34 @@ def executeMCPOnlyTestSteps(self, iteration_content, domain_proxy_objects):
     else:
       time.sleep(15)
 
+    protected_entity_records = iteration_content['protectedEntities']
+    # Creating FAD Object and Pull FAD records from SAS UUT
+    fad_test_harnesses_objects = []
+    sas_test_harness_objects = []
+    if iteration_content['sasTestHarnessData']:
+      fad_uut_object = getFullActivityDumpSasUut(self._sas,
+                                              self._sas_admin)
+    # Pull FAD from SAS Test Harnesses and fad objects are created for each SAS Test Harnesses
+    for test_harness in sas_test_harness_objects:
+      fad_test_harnesses_objects.append(getFullActivityDumpSasTestHarness(
+                           test_harness.getSasTestHarnessInterface()))
+
     # Step 25,26,27 and 28 :
     # Send heartbeat request for the grants, relinquish the grant, grant request and heartbeat for new grant
     for domain_proxy_object in domain_proxy_objects:
       domain_proxy_object.performHeartbeatAndUpdateGrants()
-
-      # Step 29
-      # TODO: Aggregate interference for DPA
-
-      # Step 30:Invoke Aggregate Interference Model
       cbsds.append(domain_proxy_object.getCbsdsWithAtLeastOneAuthorizedGrant())
 
-    # Invoke Aggregate Interference Model
-    # TODO: To invoke Aggregate Interference Model
+    cbsd_records = getFullActivityDumpSasTestHarness(cbsds).getCbsdRecords()
+    # Call Pre IAP model
+    pre_iap_filtering.preIapReferenceModel(protected_entity_records,
+                                       fad_uut_object, fad_test_harnesses_objects)
```

Comment:
IAP reference value check is required in the test case. can you be more specific what do you mean ?

Reply:
Taran and I discussed this in detail yesterday: he is right, there should not be another call to IAP here. It's a little confusing because our code structure doesn't match the test spec perfectly, but maybe this will help:

- We pull the FAD in step 15 and use it in step 17.
- Step 17 should execute the pre-IAP filtering ref model and the IAP ref model, using the FADs as input. These are the calculations that lead to A_ref.
- Because of the way the IAP ref model code is written, its output is actually what's described in the second part of step 22 (A_ref). We don't need to store or use the intermediate information, so we just go straight to calculating the A_ref value. This just means there's less to do in step 22.
- In step 30, it is not necessary to recalculate A_ref (the values from step 22 -- which were actually calculated in step 17 -- remain valid, as noted in the test spec). Since we do not need to recalculate A_ref, we don't need to pull FAD, do pre-IAP filtering, or execute the IAP ref model again (we only did those before because it was necessary to calculate A_ref). We just need to calculate the actual aggregate interference and, in the CHECK step, compare that with the earlier A_ref.

Does that help?

Reply:
Have added code for storing and then comparing the iap values with aggregate value